### PR TITLE
Add CMAKE install target

### PR DESCRIPTION
**Pull request by martin-frbg**
Add CMAKE install target (copied from a patch provided by PrimarchOfTheSpaceWolves in #957)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -198,3 +205,22 @@ set_target_properties(${OpenBLAS_LIBNAME} PROPERTIES
...
+FILE(GLOB_RECURSE INCLUDE_FILES "${CMAKE_CURRENT_SOURCE_DIR}/*.h")
+install (FILES ${INCLUDE_FILES} DESTINATION include)
+ 
+if(NOT MSVC)
+	set(DEBUG_POSTFIX "")
+	if(BUILD_DEBUG)
+		set(DEBUG_POSTFIX "_d")
+	endif()
+	install (TARGETS ${OpenBLAS_LIBNAME}_static${DEBUG_POSTFIX} DESTINATION lib)
+endif()
```
Comment (xantares) [inline CMakeLists.txt]: DEBUG_POSTFIX is already defined earlier, the install could be moved there to avoid duplicate code

File: CMakeLists.txt [hunk 2]
```diff
@@ -31,9 +31,10 @@ set(NO_LAPACKE 1)
 endif()
 
 if(BUILD_DEBUG)
-set(CMAKE_BUILD_TYPE Debug)
+	set(CMAKE_BUILD_TYPE Debug)
+	set(OpenBLAS_LIBNAME "${OpenBLAS_LIBNAME}_d")
 else()
-set(CMAKE_BUILD_TYPE Release)
+	set(CMAKE_BUILD_TYPE Release)
 endif()
```
Comment (martin-frbg) [inline CMakeLists.txt]: That is xianyi's code from 7ac7e147d, the patch only changed the indentation here.
I take it this evaluation of DEBUG_BUILD should be removed completely ?

Comment (martin-frbg) [inline CMakeLists.txt]: More to the point, should this block become something like
if( NOT CMAKE_BUILD_TYPE )
  set( CMAKE_BUILD_TYPE Release)
endif()
if (CMAKE_BUILD_TYPE = Debug)
 set (DEBUG_POSTFIX "_d")
 set(OpenBLAS_LIBNAME "${OpenBLAS_LIBNAME}_d")
else()
 set DEBUG_POSTFIX ""
endif()
which would also consolidate the definitions of DEBUG_POSTFIX as in your other comment ?

**Comment by martin-frbg** [comment]
I have reworked the PR now to query BUILD_DEBUG only if CMAKE_BUILD_TYPE is not set, and moved the conditional appending of a debug prefix to use the cmake-provided setting.
(Apologies again for the mess of incremental commits to the same file, I was working via the web interface and do not see a means of squashing the commit history there)

**Pull request merged**